test/services/stepfunctions/RegularContributionsClientTest.scala

```diff
@@ -0,0 +1,50 @@
+package services.stepfunctions
+
+import com.amazonaws.AmazonServiceException
+import com.amazonaws.services.stepfunctions.model.StateExitedEventDetails
+import com.gu.support.workers.model.CheckoutFailureReasons.CheckoutFailureReason
+import com.gu.support.workers.model.states.CheckoutFailureState
+import com.gu.support.workers.model.{CheckoutFailureReasons, Status, User}
+import org.scalatest.{FlatSpec, Matchers}
+import org.scalatest.mockito.MockitoSugar
+import org.mockito.Mockito._
+import codecs.CirceDecoders.checkoutFailureStateCodec
+import StepFunctionExecutionStatus._
+import scala.util.{Failure, Success}
+
+object StatusResults {
+  val success = StatusResponse(Status.Success, "tracking123", None)
+  def failure(reason: CheckoutFailureReason): StatusResponse = StatusResponse(Status.Failure, "tracking123", Some(reason))
+  val pending = StatusResponse(Status.Pending, "tracking123", None)
+}
+
+class RegularContributionsClientTest extends FlatSpec with Matchers with MockitoSugar {
```

Comment:
👍 thanks for adding this test. I looked at it first to see what the polling logic is supposed to do.